src/iexamine.cpp

```diff
@@ -4167,9 +4167,10 @@ void iexamine::autodoc( player &p, const tripoint &examp )
                 return;
             }
 
-            const int weight = units::to_kilogram( g->u.bodyweight() ) / 10;
+            const int weight = units::to_kilogram( patient.bodyweight() ) / 10;
+            const int surgery_duration = itemtype->bionic->difficulty * 2;
             const requirement_data req_anesth = *requirement_id( "anesthetic" ) *
-                                                itemtype->bionic->difficulty * 2 * weight;
+                                                surgery_duration * 2 * weight;
 
             if( patient.can_install_bionics( ( *itemtype ), installer, true ) ) {
                 const time_duration duration = itemtype->bionic->difficulty * 20_minutes;
```

Comment:
Btw. you have the `* 2` twice now: once in the duration calculation and once in the requirement calculation.

Reply:
One is a time_duration and the other a int, should I merge them?

Reply:
Neither. You just doubled the required amount: `*2` from calculating `surgery_duration` and **another** `*2` in the calculation of the requirement. I assumed it is not intended.

Reply:
Ah ok, I missde that. Thanks